### DISCUSSION TITLE
Upgrading on HASS-OS

### DISCUSSION THREAD

**Discussion by MrGibbage** [TD22057/insteon-mqtt#371] [Q&A]
For the longest time, I have had a problem with my garage light. My Home Assistant RPi is in the garage, so it's good and close. But still, it regularly take five seconds or so to come on. I even replaced the switch with a brand new device, but I still have the same problem. Weird. Here's my original thread about this:
https://github.com/TD22057/insteon-mqtt/issues/242

I am using HASS-OS and currently on 0.7.4, and I have auto-update checked. I think question number one is how do I upgrade to a newer version? And if anyone has any ideas why I would be having problems with just this one location, I'd appreciate their thoughts :)

Skip

**Comment by krkeegan** [comment]
Something is causing interference with the garage light receiving messages.  If you have replaced the device, the only other idea is to add other nearby devices to increase the signal strength.

Auto updates is living dangerously.  Breaking changes can happen and will be documented in the changelog, I usually like to look at those first.  This library is not likely to have security updates often if ever, which is really the only reason to need an auto-update.

I suspect it isn't working because you installed the plug in using SSH and the `addon` folder per the older method.  If you follow the instructions [here](https://github.com/TD22057/insteon-mqtt/blob/master/docs/HA_Addon_Instructions.md) it should work.

**Reply by MrGibbage** [reply]
This is getting weirder and weirder. My RPi (running HA) and the PLM are plugged into a power strip on top of my refrigerator four feet and direct line of sight with the switch. But I am definitely sure the problem is not with the switch. I created a scene with an appliancelinc and plugged it into the same powerstrip. When I try turning that scene on, there's a delay with that appliancelinc too. So is the problem with the PLM maybe? It's only a few months old, so I really hope not.

**Reply by krkeegan** [reply]
Yeah, it sounds like there is some noise in the power line around there.  Having the refrigerator right next to the PLM may not be ideal.  It could also be that the powerstrip isn't doing you any favors and the choke on it is interfering with the powerline signals.  An oscilliscope might show if there is noise on the line.

But you can also try moving the PLM to a different location and see what happens.

**Reply by MrGibbage** [reply]
I have tried three other outlets in the house. I still have the same problem. Even with the PLM and appliancelinc in the same "spider extension cord", still have the same delay. Pretty consistently too, but not quite 100% of the time. I even tried unplugging the fridge, with no luck. I'm not sure the fridge could have been the problem because it's not like the compressor is running that much, but I do see where you are going with this. I do have an O-Scope, so I may look at that too. 

Not sure if I mentioned it before, but I am trying to automate the garage light to come on when the interior door is opened (and to turn off a few minutes after closing). So the delay is very noticeable when the door is opened and the lights don't come on right away. I don't have much of my automation tied to other individual lights triggering off of doors opening, so I think I will try that. Since I have replaced the garage switch and still see the same symptoms, my hypothesis is that all of my lights will have the same issue.

Any other thoughts?

**Reply by krkeegan** [reply]
Yeah, I get it, delays like this are not helpful.

The issue is that the device is not receiving the first message.  If the delay is 5 seconds, it means the second message is received, if the delay is 10 seconds, it means the third message....  The reason the subsequent messages are received is the hop count is incremented +1 each time.

The solution, and I have been dragging my feet on it because I am not super excited about it, is to allow setting the minimum hop count on a per device basis.  This has a few downsides:
1. It requires a significant amount of code writing as anything that ingests the device config has to be fixed.
2. It will make the config.yaml file a bit more messy for users to work with, which I fear will cause more user issues
3. Setting the minimum_hops too high can cause bad hard to diagnose issues on the insteon network, which again may lead to more user issues.

Let me look at the issue, this afternoon.

**Reply by krkeegan** [reply]
OK, if you are able to run a branch, try out this branch https://github.com/krkeegan/insteon-mqtt/tree/Min_Hops

There is no documentation. But basically amend your device entry in the config file as follows:
```yaml
devices:
  dimmer:
    - 11.22.33: "device that turns on slowly name"
      min_hops: 1
```

Note that there is no dash before the min_hops.  Try it with the setting of 1 first.  If that doesn't work you can increment up by one to a maximum of 3.

**Reply by MrGibbage** [reply]
Wow, you got that done FAST ! :)

I am using HASS-OS on a RPi. I don't know if I can or cannot run a branch, but I sure would like to try!  Could you kindly tell me how I would do that?

Thanks so much for the help!!!

**Reply by krkeegan** [reply]
Yeah much of the work was actually duplicative code.  Writing the documentation, examples, and unit tests will take a bit more time.

If you are comfortable with SSH, and git, then yes you can test this out.  I wrote up some sparse instructions [here](https://github.com/TD22057/insteon-mqtt/blob/master/docs/hassio_development.md) that would point you in the right direction.  But they are not a recipe you can just copy, you would have to at least vaguely understand git to get the proper repository and branch.

Otherwise, it will probably be about a week to finish up the changes, test them out, and push a new release.

**Reply by philletourneau** [reply]
Just piggybacking on this, but I have auto-update on HASS as well and I'm still running 0.7.2. Is upgrading broken or requires manual intervention? @krkeegan

**Comment by krkeegan** [comment]
OK, there is an option for `min_hops` now.  See [the documentation](https://github.com/TD22057/insteon-mqtt/blob/master/docs/config_extra.md) for guidance on how to use it.

Let me know if this solves your delay issue.

**Reply by MrGibbage** [reply]
Just upgraded. Sadly, still have the same problem. But I am starting to wonder if it is an insteon-mqtt problem at all. Here's what I did today to troubleshoot. Again, as a reminder, what I am trying to do is have my garage light come on after I open the door. I have an Elk security system which is integrated with HASS and that provides the trigger. Today for troubleshooting, I mixed things up. I have a kitchen light that is close to the garage door, and I have a back door which is close to the kitchen and garage as well. I wanted to see what would happen if I had the garage door trigger the kitchen light and I wanted to see what would happen if the back door triggered those lights. So I have garage door->garage light, garage door->kitchen light, back door->garage light and back door->kitchen light. Bottom line, it works perfectly when the back door is the trigger. HOWEVER, I can see the insteon-mqtt and HASS logs both show the light getting triggered immediately when the garage door  is opened, so I can't think of anything that could be causing the issue. So, with all of that being said, here are some logs to look over.

I also have one other door that triggers a light, and it works perfectly 100 percent of the time. So it is something with that door, but I don't know what.

I do have the min_hops: 1 setting for the garage light.

````
Kitchen responder to garage door opening
2021-04-09 18:01:46.516 INFO Protocol: Read 0x50: Std: 3b.ac.52 Type.ALL_LINK_BROADCAST mh:3 hl:3 grp: 01 cmd: 13 00
2021-04-09 18:01:46.517 DEBUG Protocol: Setting next write time: 18:01:46.777
2021-04-09 18:01:46.518 DEBUG MsgHistory: Received 0 hops, total 0 for 9 entries
2021-04-09 18:01:46.519 INFO Broadcast: Handling all link broadcast for 3b.ac.52 'eat-in kitchen hanging light'
2021-04-09 18:01:46.520 DEBUG Protocol: Setting next write time: 18:01:47.020
2021-04-09 18:01:46.521 INFO Base: Device 3b.ac.52 broadcast grp: 1 on: False mode: normal
2021-04-09 18:01:46.522 INFO Base: Setting device 3b.ac.52 (eat-in kitchen hanging light) on False level 0 normal device
2021-04-09 18:01:46.524 INFO StateTopic: MQTT received state 3b.ac.52 (eat-in kitchen hanging light) on: {'is_on': False, 'level': 0, 'mode': <Mode.NORMAL: 'normal'>, 'button': 1, 'reason': 'device'}
2021-04-09 18:01:46.526 DEBUG Mqtt: Sending PUBLISH (d0, q1, r1, m142), 'b'insteon/3b.ac.52/state'', ... (37 bytes)
2021-04-09 18:01:46.529 DEBUG Mqtt: MQTT publish insteon/3b.ac.52/state { "state" : "OFF", "brightness" : 0 } qos=1 ret=1
2021-04-09 18:01:46.530 DEBUG Base: Found 0 responders in group 1
2021-04-09 18:01:46.531 DEBUG Base: Group 1 -> []
2021-04-09 18:01:46.532 DEBUG Mqtt: MQTT writing
2021-04-09 18:01:46.710 DEBUG Mqtt: Received PUBACK (Mid: 142)
2021-04-09 18:01:46.788 INFO Protocol: Read 0x50: Std: 3b.ac.52 Type.ALL_LINK_CLEANUP mh:0 hl:0 grp: 01 cmd: 13 01
2021-04-09 18:01:46.789 DEBUG MsgHistory: Received 0 hops, total 0 for 10 entries
2021-04-09 18:01:46.900 INFO Protocol: Read 0x50: Std: 3b.ac.52 Type.ALL_LINK_BROADCAST mh:3 hl:3 grp: 01 cmd: 06 00
2021-04-09 18:01:46.902 DEBUG Protocol: Setting next write time: 18:01:47.161
2021-04-09 18:01:46.903 DEBUG MsgHistory: Received 0 hops, total 0 for 11 entries
2021-04-09 18:01:46.905 DEBUG Protocol: Setting next write time: 18:01:46.905
2021-04-09 18:01:46.907 DEBUG Protocol: Setting next write time: 18:01:47.161
2021-04-09 18:01:46.908 DEBUG Broadcast: Cleanup report for 3b.ac.52, grp 1 success.
2021-04-09 18:01:51.628 DEBUG Mqtt: Received PUBLISH (d0, q1, r0, m9158), 'insteon/command/3b.ac.52', ...  (12 bytes)
2021-04-09 18:01:51.630 DEBUG Mqtt: Sending PUBACK (Mid: 9158)
2021-04-09 18:01:51.631 INFO Mqtt: MQTT message insteon/command/3b.ac.52 b'{"cmd":"on"}'
2021-04-09 18:01:51.632 UI Mqtt: Commanding dimmer device 3b.ac.52 (eat-in kitchen hanging light) cmd=on
2021-04-09 18:01:51.633 INFO ResponderBase: Device 3b.ac.52 grp: 1 cmd: on normal
2021-04-09 18:01:51.633 DEBUG MsgHistory: Average hops 0.0, using 0
2021-04-09 18:01:51.634 INFO Protocol: Write message to modem: Std: 3b.ac.52, Type.DIRECT mh:0 hl:0, 11 ff 
2021-04-09 18:01:51.635 DEBUG Protocol: Write bytes to modem: 02623bac520011ff
2021-04-09 18:01:51.636 DEBUG Serial: Wrote 8 bytes to serial /dev/ttyUSB0
2021-04-09 18:01:51.650 INFO Protocol: PLM is busy, pausing briefly
2021-04-09 18:01:51.651 DEBUG Protocol: Setting next write time: 18:01:51.951
2021-04-09 18:01:57.658 WARNING Base: Handler timed out 1 of 3 sent: Std: 3b.ac.52, Type.DIRECT mh:0 hl:0, 11 ff 
2021-04-09 18:01:57.660 DEBUG Base: Increasing max_hops to 1
2021-04-09 18:01:57.662 INFO Protocol: Write message to modem: Std: 3b.ac.52, Type.DIRECT mh:1 hl:1, 11 ff 
2021-04-09 18:01:57.663 DEBUG Protocol: Write bytes to modem: 02623bac520511ff
2021-04-09 18:01:57.665 DEBUG Serial: Wrote 8 bytes to serial /dev/ttyUSB0
2021-04-09 18:01:57.681 INFO Protocol: PLM is busy, pausing briefly
2021-04-09 18:01:57.682 DEBUG Protocol: Setting next write time: 18:01:57.982
2021-04-09 18:01:57.683 INFO Protocol: Read 0x62: Std: 3b.ac.52, Type.DIRECT mh:1 hl:1, 11 ff ack: True
2021-04-09 18:01:57.684 DEBUG Protocol: Passing msg to write handler: StandardCmd handler
2021-04-09 18:01:57.685 DEBUG StandardCmd: 3b.ac.52 got PLM ACK
2021-04-09 18:01:57.890 INFO Protocol: Read 0x50: Std: 3b.ac.52->53.c1.0e Type.DIRECT_ACK mh:1 hl:1 cmd: 11 ff
2021-04-09 18:01:57.891 DEBUG MsgHistory: Received 0 hops, total 0 for 11 entries
2021-04-09 18:01:57.893 DEBUG Protocol: Passing msg to write handler: StandardCmd handler
2021-04-09 18:01:57.894 DEBUG ResponderBase: Device 3b.ac.52 ACK: Std: 3b.ac.52->53.c1.0e Type.DIRECT_ACK mh:1 hl:1 cmd: 11 ff
2021-04-09 18:01:57.895 INFO Base: Setting device 3b.ac.52 (eat-in kitchen hanging light) on True level 255 normal command
2021-04-09 18:01:57.896 INFO StateTopic: MQTT received state 3b.ac.52 (eat-in kitchen hanging light) on: {'is_on': True, 'level': 255, 'mode': <Mode.NORMAL: 'normal'>, 'button': None, 'reason': 'command'}
2021-04-09 18:01:57.898 DEBUG Mqtt: Sending PUBLISH (d0, q1, r1, m143), 'b'insteon/3b.ac.52/state'', ... (38 bytes)
2021-04-09 18:01:57.900 DEBUG Mqtt: MQTT publish insteon/3b.ac.52/state { "state" : "ON", "brightness" : 255 } qos=1 ret=1
2021-04-09 18:01:57.901 UI Mqtt: Device state updated to on=True
2021-04-09 18:01:57.902 DEBUG Protocol: Write handler finished
2021-04-09 18:01:57.903 DEBUG Mqtt: MQTT writing
2021-04-09 18:01:57.905 DEBUG Mqtt: Received PUBACK (Mid: 143)

Garage responder to garage door opening
2021-04-09 18:06:19.073 DEBUG Mqtt: Received PUBLISH (d0, q1, r0, m9160), 'insteon/command/50.d0.3a', ...  (12 bytes)
2021-04-09 18:06:19.074 DEBUG Mqtt: Sending PUBACK (Mid: 9160)
2021-04-09 18:06:19.076 INFO Mqtt: MQTT message insteon/command/50.d0.3a b'{"cmd":"on"}'
2021-04-09 18:06:19.078 UI Mqtt: Commanding switch device 50.d0.3a (garage) cmd=on
2021-04-09 18:06:19.079 INFO ResponderBase: Device 50.d0.3a grp: 1 cmd: on normal
2021-04-09 18:06:19.080 DEBUG MsgHistory: Average hops 0.4, using 1
2021-04-09 18:06:19.081 INFO Protocol: Write message to modem: Std: 50.d0.3a, Type.DIRECT mh:1 hl:1, 11 ff 
2021-04-09 18:06:19.081 DEBUG Protocol: Write bytes to modem: 026250d03a0511ff
2021-04-09 18:06:19.082 DEBUG Serial: Wrote 8 bytes to serial /dev/ttyUSB0
2021-04-09 18:06:19.109 INFO Protocol: PLM is busy, pausing briefly
2021-04-09 18:06:19.110 DEBUG Protocol: Setting next write time: 18:06:19.410
2021-04-09 18:06:25.116 WARNING Base: Handler timed out 1 of 3 sent: Std: 50.d0.3a, Type.DIRECT mh:1 hl:1, 11 ff 
2021-04-09 18:06:25.118 DEBUG Base: Increasing max_hops to 2
2021-04-09 18:06:25.121 INFO Protocol: Write message to modem: Std: 50.d0.3a, Type.DIRECT mh:2 hl:2, 11 ff 
2021-04-09 18:06:25.122 DEBUG Protocol: Write bytes to modem: 026250d03a0a11ff
2021-04-09 18:06:25.124 DEBUG Serial: Wrote 8 bytes to serial /dev/ttyUSB0
2021-04-09 18:06:25.140 INFO Protocol: PLM is busy, pausing briefly
2021-04-09 18:06:25.142 DEBUG Protocol: Setting next write time: 18:06:25.441
2021-04-09 18:06:25.143 INFO Protocol: Read 0x62: Std: 50.d0.3a, Type.DIRECT mh:2 hl:2, 11 ff ack: True
2021-04-09 18:06:25.144 DEBUG Protocol: Passing msg to write handler: StandardCmd handler
2021-04-09 18:06:25.146 DEBUG StandardCmd: 50.d0.3a got PLM ACK
2021-04-09 18:06:25.396 INFO Protocol: Read 0x50: Std: 50.d0.3a->53.c1.0e Type.DIRECT_ACK mh:2 hl:2 cmd: 11 ff
2021-04-09 18:06:25.398 DEBUG Protocol: Setting next write time: 18:06:25.570
2021-04-09 18:06:25.399 DEBUG MsgHistory: Received 0 hops, total 4 for 11 entries
2021-04-09 18:06:25.400 DEBUG Protocol: Passing msg to write handler: StandardCmd handler
2021-04-09 18:06:25.401 DEBUG ResponderBase: Device 50.d0.3a ACK: Std: 50.d0.3a->53.c1.0e Type.DIRECT_ACK mh:2 hl:2 cmd: 11 ff
2021-04-09 18:06:25.402 INFO Base: Setting device 50.d0.3a (garage) on True level 255 normal command
2021-04-09 18:06:25.404 INFO StateTopic: MQTT received state 50.d0.3a (garage) on: {'is_on': True, 'level': 255, 'mode': <Mode.NORMAL: 'normal'>, 'button': None, 'reason': 'command'}
2021-04-09 18:06:25.405 DEBUG Mqtt: Sending PUBLISH (d0, q1, r1, m144), 'b'insteon/50.d0.3a/state'', ... (2 bytes)
2021-04-09 18:06:25.407 DEBUG Mqtt: MQTT publish insteon/50.d0.3a/state ON qos=1 ret=1
2021-04-09 18:06:25.407 UI Mqtt: Device state updated to on=True
2021-04-09 18:06:25.409 DEBUG Protocol: Write handler finished
2021-04-09 18:06:25.410 DEBUG Mqtt: MQTT writing
2021-04-09 18:06:25.585 DEBUG Mqtt: Received PUBACK (Mid: 144)

Kitchen Responder to back door
2021-04-09 18:08:52.266 DEBUG Mqtt: Received PUBLISH (d0, q1, r0, m9161), 'insteon/command/3b.ac.52', ...  (12 bytes)
2021-04-09 18:08:52.268 DEBUG Mqtt: Sending PUBACK (Mid: 9161)
2021-04-09 18:08:52.270 INFO Mqtt: MQTT message insteon/command/3b.ac.52 b'{"cmd":"on"}'
2021-04-09 18:08:52.271 UI Mqtt: Commanding dimmer device 3b.ac.52 (eat-in kitchen hanging light) cmd=on
2021-04-09 18:08:52.272 INFO ResponderBase: Device 3b.ac.52 grp: 1 cmd: on normal
2021-04-09 18:08:52.273 DEBUG MsgHistory: Average hops 0.0, using 0
2021-04-09 18:08:52.274 INFO Protocol: Write message to modem: Std: 3b.ac.52, Type.DIRECT mh:0 hl:0, 11 ff 
2021-04-09 18:08:52.275 DEBUG Protocol: Write bytes to modem: 02623bac520011ff
2021-04-09 18:08:52.277 DEBUG Serial: Wrote 8 bytes to serial /dev/ttyUSB0
2021-04-09 18:08:52.299 INFO Protocol: Read 0x62: Std: 3b.ac.52, Type.DIRECT mh:0 hl:0, 11 ff ack: True
2021-04-09 18:08:52.301 DEBUG Protocol: Passing msg to write handler: StandardCmd handler
2021-04-09 18:08:52.302 DEBUG StandardCmd: 3b.ac.52 got PLM ACK
2021-04-09 18:08:52.443 INFO Protocol: Read 0x50: Std: 3b.ac.52->53.c1.0e Type.DIRECT_ACK mh:0 hl:0 cmd: 11 ff
2021-04-09 18:08:52.445 DEBUG Protocol: Setting next write time: 18:08:52.443
2021-04-09 18:08:52.446 DEBUG MsgHistory: Received 0 hops, total 0 for 11 entries
2021-04-09 18:08:52.447 DEBUG Protocol: Passing msg to write handler: StandardCmd handler
2021-04-09 18:08:52.449 DEBUG ResponderBase: Device 3b.ac.52 ACK: Std: 3b.ac.52->53.c1.0e Type.DIRECT_ACK mh:0 hl:0 cmd: 11 ff
2021-04-09 18:08:52.450 INFO Base: Setting device 3b.ac.52 (eat-in kitchen hanging light) on True level 255 normal command
2021-04-09 18:08:52.451 INFO StateTopic: MQTT received state 3b.ac.52 (eat-in kitchen hanging light) on: {'is_on': True, 'level': 255, 'mode': <Mode.NORMAL: 'normal'>, 'button': None, 'reason': 'command'}
2021-04-09 18:08:52.453 DEBUG Mqtt: Sending PUBLISH (d0, q1, r1, m146), 'b'insteon/3b.ac.52/state'', ... (38 bytes)
2021-04-09 18:08:52.455 DEBUG Mqtt: MQTT publish insteon/3b.ac.52/state { "state" : "ON", "brightness" : 255 } qos=1 ret=1
2021-04-09 18:08:52.456 UI Mqtt: Device state updated to on=True
2021-04-09 18:08:52.457 DEBUG Protocol: Write handler finished
2021-04-09 18:08:52.459 DEBUG Mqtt: Received PUBACK (Mid: 146)
2021-04-09 18:08:52.460 DEBUG Mqtt: MQTT writing

Garage Responder to back door
2021-04-09 18:13:45.671 DEBUG Mqtt: Received PUBLISH (d0, q1, r0, m9167), 'insteon/command/50.d0.3a', ...  (12 bytes)
2021-04-09 18:13:45.672 DEBUG Mqtt: Sending PUBACK (Mid: 9167)
2021-04-09 18:13:45.673 INFO Mqtt: MQTT message insteon/command/50.d0.3a b'{"cmd":"on"}'
2021-04-09 18:13:45.674 UI Mqtt: Commanding switch device 50.d0.3a (garage) cmd=on
2021-04-09 18:13:45.675 INFO ResponderBase: Device 50.d0.3a grp: 1 cmd: on normal
2021-04-09 18:13:45.676 DEBUG MsgHistory: Average hops 0.2, using 1
2021-04-09 18:13:45.677 INFO Protocol: Write message to modem: Std: 50.d0.3a, Type.DIRECT mh:1 hl:1, 11 ff 
2021-04-09 18:13:45.677 DEBUG Protocol: Write bytes to modem: 026250d03a0511ff
2021-04-09 18:13:45.678 DEBUG Serial: Wrote 8 bytes to serial /dev/ttyUSB0
2021-04-09 18:13:45.688 INFO Protocol: Read 0x62: Std: 50.d0.3a, Type.DIRECT mh:1 hl:1, 11 ff ack: True
2021-04-09 18:13:45.689 DEBUG Protocol: Passing msg to write handler: StandardCmd handler
2021-04-09 18:13:45.690 DEBUG StandardCmd: 50.d0.3a got PLM ACK
2021-04-09 18:13:45.897 INFO Protocol: Read 0x50: Std: 50.d0.3a->53.c1.0e Type.DIRECT_ACK mh:1 hl:1 cmd: 11 ff
2021-04-09 18:13:45.898 DEBUG Protocol: Setting next write time: 18:13:45.984
2021-04-09 18:13:45.900 DEBUG MsgHistory: Received 0 hops, total 2 for 11 entries
2021-04-09 18:13:45.901 DEBUG Protocol: Passing msg to write handler: StandardCmd handler
2021-04-09 18:13:45.902 DEBUG ResponderBase: Device 50.d0.3a ACK: Std: 50.d0.3a->53.c1.0e Type.DIRECT_ACK mh:1 hl:1 cmd: 11 ff
2021-04-09 18:13:45.904 INFO Base: Setting device 50.d0.3a (garage) on True level 255 normal command
2021-04-09 18:13:45.905 INFO StateTopic: MQTT received state 50.d0.3a (garage) on: {'is_on': True, 'level': 255, 'mode': <Mode.NORMAL: 'normal'>, 'button': None, 'reason': 'command'}
2021-04-09 18:13:45.906 DEBUG Mqtt: Sending PUBLISH (d0, q1, r1, m152), 'b'insteon/50.d0.3a/state'', ... (2 bytes)
2021-04-09 18:13:45.908 DEBUG Mqtt: MQTT publish insteon/50.d0.3a/state ON qos=1 ret=1
2021-04-09 18:13:45.909 UI Mqtt: Device state updated to on=True
2021-04-09 18:13:45.910 DEBUG Protocol: Write handler finished
2021-04-09 18:13:45.912 DEBUG Mqtt: Received PUBACK (Mid: 152)
2021-04-09 18:13:45.913 DEBUG Mqtt: MQTT writing
````

**Reply by krkeegan** [reply]
Don't take any offence to this, but it seems to be hard to keep you focused on one problem.  I feel like every response always takes a slightly different direction each time with new variables and multiple problems.

It is hard for me to tell what exactly you think the "problem" is in all of these cases.  But I think you are saying that sometimes things take 5 seconds and sometimes they don't.  I don't think the item triggering the light to change matters.  If I am reading between the lines well enough, I think in each of these scenarios you describe 
>garage door->garage light, garage door->kitchen light, back door->garage light and back door->kitchen light

HomeAssistant is in the middle handling the automation.  Is that correct?  I looks to me like each of the automations trigger a call to turn on the `garage light` or `kitchen light`.  If this is the case, InsteonMQTT is oblivious to what triggered the automation to run.

Taking the issues I see one at a time:

- Garage light - Let's try increasing the `min_hops: 2` as the setting.  In the section you have labeled `Garage responder to garage door opening` the message doesn't seem to reach it with only 1 hop.
- Kitchen light - if you want to try setting `min_hops: 1` for this device it may help.

These log details:

> kitchen light
```
2021-04-09 18:01:57.658 WARNING Base: Handler timed out 1 of 3 sent: Std: 3b.ac.52, Type.DIRECT mh:0 hl:0, 11 ff 
2021-04-09 18:01:57.660 DEBUG Base: Increasing max_hops to 1
```

and 

> garage light
```
2021-04-09 18:06:25.116 WARNING Base: Handler timed out 1 of 3 sent: Std: 50.d0.3a, Type.DIRECT mh:1 hl:1, 11 ff 
2021-04-09 18:06:25.118 DEBUG Base: Increasing max_hops to 2
```

Are telling you that the initial command sent to these devices did not make it to the device.  The subsequent line tells you that the hops had to be increased.  For the `garage light` it required 2 hops and the `kitchen light` 1.  If you look at this messaging you might have been able to solve your own issue with the garage light by seeing that it required 2 hops.

Messages like this may happen from time to time.  We know from experience that the `garage light` has been an issue, so setting the `min_hops` for it makes sense.  I don't know if the `kitchen light` has always been an issue.  If it has then the `min_hops` setting for it makes sense as well.  But it may only be acting up because you just restarted InsteonMQTT and may not need the `min_hops` setting.

Bottom line, and I know this sounds like I get a commission from Insteon, but my guess is that you just don't have a high enough concentration of Insteon devices to have a consistent mesh network working all the time.
```

**Reply by MrGibbage** [reply]
First of all, absolutely no offense taken. I have been troubleshooting Navy electronic systems for many years, and I really felt I could get a handle on this. The reason I went down this different path was because I realized that the other automations in my house were responding perfectly. Even when I add the garage light to that automation, just as a test. I was starting to think it was more than just bad insteon comms. So I tried the other things. Again, no offense taken.

So, I tried increasing to min_hops: 2 and ran it again. And I saw the debug message that it was increasing max_hops to 3. So I increased it to three, and now it still says increasing max_hops to 3. That surprised me, so I restarted the insteon-mqtt and I still get the same messages.

As for the number of devices in that area, the PLM is still sitting on top of the refrigerator, but I have unplugged it while testing. It is three feet, direct line of sight to the switch. The PLM is less than six months old, and the garage switch is dual band. The switch in in a dual gang box with an insteon dimmer (the garage light is an on-off switch). And I have a dual band appliancelinc in line of sight. The PLM and these devices are all on the same circuit in my house. Four devices within a six foot diameter circle, all line of sight. Do you really think I might need more in there? I do have one more dual band appliancelinc dimmer that I can put in there.

Here's the latest log, with min_hops set to 3:
````
2021-04-11 12:15:19.597 DEBUG Mqtt: Received PUBLISH (d0, q1, r0, m9399), 'insteon/command/50.d0.3a', ...  (12 bytes)
2021-04-11 12:15:19.600 DEBUG Mqtt: Sending PUBACK (Mid: 9399)
2021-04-11 12:15:19.601 INFO Mqtt: MQTT message insteon/command/50.d0.3a b'{"cmd":"on"}'
2021-04-11 12:15:19.602 UI Mqtt: Commanding switch device 50.d0.3a (garage) cmd=on
2021-04-11 12:15:19.603 INFO ResponderBase: Device 50.d0.3a grp: 1 cmd: on normal
2021-04-11 12:15:19.603 DEBUG MsgHistory: Average hops 0.0, using 0
2021-04-11 12:15:19.604 INFO Protocol: Write message to modem: Std: 50.d0.3a, Type.DIRECT mh:3 hl:3, 11 ff 
2021-04-11 12:15:19.605 DEBUG Protocol: Write bytes to modem: 026250d03a0f11ff
2021-04-11 12:15:19.606 DEBUG Serial: Wrote 8 bytes to serial /dev/ttyUSB0
2021-04-11 12:15:25.632 WARNING Base: Handler timed out 1 of 3 sent: Std: 50.d0.3a, Type.DIRECT mh:3 hl:3, 11 ff 
2021-04-11 12:15:25.634 DEBUG Base: Increasing max_hops to 3
2021-04-11 12:15:25.637 INFO Protocol: Write message to modem: Std: 50.d0.3a, Type.DIRECT mh:3 hl:3, 11 ff 
2021-04-11 12:15:25.638 DEBUG Protocol: Write bytes to modem: 026250d03a0f11ff
2021-04-11 12:15:25.639 DEBUG Serial: Wrote 8 bytes to serial /dev/ttyUSB0
2021-04-11 12:15:25.641 INFO Protocol: PLM is busy, pausing briefly
2021-04-11 12:15:25.642 DEBUG Protocol: Setting next write time: 12:15:25.942
2021-04-11 12:15:25.657 INFO Protocol: Read 0x62: Std: 50.d0.3a, Type.DIRECT mh:3 hl:3, 11 ff ack: True
2021-04-11 12:15:25.658 DEBUG Protocol: Passing msg to write handler: StandardCmd handler
2021-04-11 12:15:25.659 DEBUG StandardCmd: 50.d0.3a got PLM ACK
2021-04-11 12:15:25.946 INFO Protocol: Read 0x50: Std: 50.d0.3a->53.c1.0e Type.DIRECT_ACK mh:3 hl:3 cmd: 11 ff
2021-04-11 12:15:25.947 DEBUG Protocol: Setting next write time: 12:15:26.207
2021-04-11 12:15:25.948 DEBUG MsgHistory: Received 0 hops, total 0 for 5 entries
2021-04-11 12:15:25.949 DEBUG Protocol: Passing msg to write handler: StandardCmd handler
2021-04-11 12:15:25.950 DEBUG ResponderBase: Device 50.d0.3a ACK: Std: 50.d0.3a->53.c1.0e Type.DIRECT_ACK mh:3 hl:3 cmd: 11 ff
2021-04-11 12:15:25.951 INFO Base: Setting device 50.d0.3a (garage) on True level 255 normal command
2021-04-11 12:15:25.953 INFO StateTopic: MQTT received state 50.d0.3a (garage) on: {'is_on': True, 'level': 255, 'mode': <Mode.NORMAL: 'normal'>, 'button': None, 'reason': 'command'}
2021-04-11 12:15:25.954 DEBUG Mqtt: Sending PUBLISH (d0, q1, r1, m129), 'b'insteon/50.d0.3a/state'', ... (2 bytes)
2021-04-11 12:15:25.957 DEBUG Mqtt: MQTT publish insteon/50.d0.3a/state ON qos=1 ret=1
2021-04-11 12:15:25.958 UI Mqtt: Device state updated to on=True
2021-04-11 12:15:25.959 DEBUG Protocol: Write handler finished
2021-04-11 12:15:25.960 DEBUG Mqtt: MQTT writing
2021-04-11 12:15:26.219 DEBUG Mqtt: Received PUBACK (Mid: 129)
````

I promise to not go down any other paths! I very much appreciate your help, and the least I can do is confuse the matter with my home grown science experiments!

**Reply by krkeegan** [reply]
Yeah, so sadly, it looks like this is a problem in Insteon network:
1. It could be the device, but you appear to have swapped it out at least once from the original one.
2. It could be the location
2a. if you have a decent amount of other devices and it is only this one, then something about that location is making it hard to hear the messages, interference in the line, a metal box blocking wireless signals, just plain far away.
2b. if you are having issues with multiple devices, it may be that the modem is in a bad location.
4. You may not have enough devices, or at least enough on or near this electrical circuit.  Check what other things on on the same circuit as this light switch.  Is there any other light or outlet you may want a smart device on?  It could help increase the signal strength for this device.

My recommendation is probably to stay with `min_hops: 3` for this device for the time being.  You may see some improvements.  Sadly that is the highest things will go.  If it seems like it is worse, try backing down to 2 or 1.  This line:

```
2021-04-11 12:15:25.946 INFO Protocol: Read 0x50: Std: 50.d0.3a->53.c1.0e Type.DIRECT_ACK mh:3 hl:3 cmd: 11 ff
```

Is a message sent from the garage device to the modem.  It has `mh:3 hl:3` meaning the message was sent with a max_hops of 3 and still has 3 hops left.  In other words, the modem was able to hear this device directly without any hops.  So, using a higher `min_hops` such as 2 or 3 may only be yelling more without much benefit.

In the longer run, I would see if it is at all possible to add another device somewhere between the modem and the garage.  A device not on the same circuit but between them could still help with wireless signals.  A device on the same circuit might be even better because it would strengthen the wired signal.  Adding a device would mean that when the additional hops occur, both the modem and the extra device would be yelling at the same time and that might help.

I don't think there is much more I can add or that we can do from our side for this device.  But I am happy to help with the next problem if you have one.

**Reply by MrGibbage** [reply]
Totally tracking what you are saying. I can make a modem scene with these three devices and see how that helps. I'll keep trying. I do appreciate your help very much! If I get it solved, I'll be sure to let you know though.

**Reply by krkeegan** [reply]
A modem scene won't do anything to increase the likelihood that the garage light will hear the message.  That said, it may make things more palatable for you, since the other lights will likely react immediately rather than getting blocked by waiting for the garage to acknowledge the command.

**Comment by krkeegan** [comment]
Just adding another data point.  I had a 2477S switch that was engine version i2CS.  A few months ago it started having issues with i2 commands.  Then most recently it started to fail 50% to on/off commands.  

I assumed the device was bad and replaced it.  But right away, I noticed that the new device was also acting funny.  

So I plugged a few mobile devices into the same electrical circuit thinking maybe it can't hear commands.  This only provided mild relief.

I turned the circuit off and identified the other electrical devices in the circuit.  I decided, somewhat arbitrarily, that a powered subwoofer and possibly a power strip (powering an audio amplifier, a directv box, a wifi access point, and a chromecast) could be causing issues.

So I plugged in a [FilterLinc](https://www.smarthome.com/products/filterlinc-10-amp-plug-in-noise-filter) to isolate the powerstrip and the subwoofer.  And poof, all the issues went away.  The new device works perfect.  Even the old device wired up in another location works perfect.

I can't yet say which device was causing the issue, but it is pretty clear that one of them was causing a lot of noise and interference.  The wireless portion of Insteon was not enough by itself to overcome the issue.

The only bummer, is it seems like Insteon may no longer be making the Filterlinc devices.  Not sure why that is, or whether other generic power filtering devices could be used instead.

Anyways, I hope this helps someone, sometimes you have to test whether non-Insteon devices are causing interference.
